akd/src/utils.rs

```diff
@@ -53,8 +54,50 @@ pub(crate) fn empty_node_hash_no_label<H: Hasher>() -> H::Digest {
     H::hash(&EMPTY_VALUE)
 }
 
-// FIXME: Make a real commitment here, alongwith a blinding factor. See issue #123
-/// Gets the bytes for a value.
-pub(crate) fn value_to_bytes<H: Hasher>(value: &crate::AkdValue) -> H::Digest {
-    H::hash(value)
+// Corresponds to the I2OSP() function from RFC8017, prepending the length of
+// a byte array to the byte array (so that it is ready for serialization and hashing)
+//
+// Input byte array cannot be > 2^64-1 in length
+pub(crate) fn i2osp_array(input: &[u8]) -> Vec<u8> {
+    [&(input.len() as u64).to_be_bytes(), input].concat()
+}
+
+// Commitment helper functions
+
+// Used by the server to produce a commitment proof for an AkdLabel, version, and AkdValue
+pub(crate) fn get_commitment_proof<H: Hasher>(
+    commitment_key: &[u8],
+    label: &AkdLabel,
+    version: u64,
+    value: &AkdValue,
+) -> H::Digest {
+    H::hash(
+        &[
+            commitment_key,
+            &i2osp_array(label),
+            &version.to_be_bytes(),
+            &i2osp_array(value),
+        ]
+        .concat(),
+    )
+}
+
+// Used by the server to produce a commitment for an AkdLabel, version, and AkdValue
+//
+// proof = H(commitment_key, label, version, value)
```

Comment:
May I suggest adding here that `proof` is a nonce used to create a hiding and binding commitment using a cryptographic hash function?

Reply:
For uniqueness, `AkdLabel` and `version` should be enough. Could this save us a few cycles?

Reply:
It could. My thinking was that by including the AkdValue, we will for sure have uniqueness, whereas perhaps without AkdValue there might be some edge case that allows for an attack. So this was "just to be safe". But if it is really not necessary, then I suppose we can omit it.

Reply:
I can't think of one atm but adding `AkdValue` would certainly create a stronger bond between the nonce and the data it is created for. Let's keep it!

Reply:
In the latest commit, adjusted the comment here to include the detail about the proof acting as a nonce.